### PR TITLE
Support groupChar and decimalChar in the number type

### PR DESCRIPTION
**Pull request by akariv**
Also made some changes to the base type (to allow getting extra properties.
    Fixed a few bugs along the way.

### CODE REVIEW

File: jsontableschema/types/base.py [hunk 1]
```diff
@@ -70,6 +71,28 @@ def __init__(self, field=None):
             self.__format_main = 'fmt'
             self.__format_fmt = self.__format.strip('fmt:')
 
+    @property
+    def _field(self):
+        """Returns original field object for this type
+        Should be used for getting extra properties for this type
+        """
+        return self.__field if self.__field is not None else {}
```
Comment (roll) [inline jsontableschema/types/base.py]: I support this logic should be moved to __init__:
```
# Set default field
self.__field = {}
```

**Comment by roll** [comment]
@akariv 
Could we do not introduce protected fields for this? Codebase is already is kinda complicated and I'm not sure we should add here other inheritance things.

So I suppose we need to move `_is_null` to private namespace and:
- get field in NumberType.__init__ (the most lean way)
- add JTSType.field public property if there is a feel it will be highly-used

**Comment by roll** [comment]
Also `field` could be passed to `cast_<format>` function. It will really decouple things.

File: jsontableschema/types/base.py (hunk 1 above)
Comment (akariv) [inline jsontableschema/types/base.py]: Agreed

**Comment by akariv** [comment]
@roll I don't think every type that needs the field should intercept the constructor and save a local copy of it. To me it feels like a service that needs to be in the base class, not the inheriting classes. 
Therefore, I'll change it as per your 2nd option.

[Aside: As for passing it in the `cast_<format>` methods - I disagree, since:
(a) Type classes are often created from the schema in one place and used with values in another. The caller definitely doesn't need to know how to parse the schema (that's what _this_ library is for)
(b) It would introduce a breaking change all users of these classes (esp. if anyone is inheriting from them), as the method signatures would need to be updated to reflect that new keyword argument.]

**Comment by roll** [comment]
[Aside: I agree with your disagree it was'n a great idea - there is some confusion because a class' acting like a `Field` (as it names in all ORMs etc) here named `Type`. But I'm not sure we should public things like `NumberType.cast_currency` etc before it will be requested - now it's not in the docs so it's more common to count it as not public API.]

**Comment by akariv** [comment]
Alright, I made the changes - please take a look @roll

**Comment by roll** [comment]
Thanks!

https://pypi.python.org/pypi/jsontableschema/0.6.2

**Pull request merged**